### ISSUE TITLE
Hotdoc can't find llvm-config with alternative names

### ISSUE THREAD

**Issue by lantw44** [hotdoc/hotdoc#177]
On FreeBSD, it is not allowed for any package to pull in unversioned `llvm-config` command as a dependency. Therefore, users usually have LLVM installed, but `llvm-config` can't be found in PATH. In meson, it has a list of known alternative names of `llvm-config`, so it can find LLVM on these systems. Can hotdoc use the same list to do the same thing, or provide a way to specify the name of `llvm-config` program?
https://github.com/mesonbuild/meson/blob/0.51.0/mesonbuild/dependencies/dev.py#L198

**Comment by MathieuDuponchelle** [comment]
yay for non-standard configuration tools :( I'll be happy to merge a commit that makes this work on your platform, possibly inspired by that meson piece of code. I don't much like it, but if that's what it takes so be it. I don't have a BSD install, so it's probably best if you implement this on your end to save us some back and forth :)

**Comment by thiblahute** [comment]
Wouldn't it be better to add a path to the tool as argument? This way meson could provide it automatically and avoid us to have to copy meson code and update it?

**Comment by lantw44** [comment]
I just checked the usage of `llvm-config` in` hotdoc/extensions/c/c_extension.py`. It looked like it just needs to know where it can find libraries and headers. Probably hotdoc can provide options for meson to specify these two directories directly. Meson is able to find LLVM with both `llvm-config` and CMake. If it decides to use CMake, it probably doesn't know where `llvm-config` is.

**Comment by MathieuDuponchelle** [comment]
how does CMake determine these paths ?

**Comment by MathieuDuponchelle** [comment]
I wouldn't mind a build-system agnostic solution fwiw, if we rely on meson to pass us this information this means the problem will still exist with other build systems

**Comment by thiblahute** [comment]
Other build systems will need to provide the info too, this would just override our basic way of figuring stuff out.

**Comment by MathieuDuponchelle** [comment]
> Other build systems will need to provide the info too, this would just override our basic way of figuring stuff out.

Yes, that's what I'm saying, perhaps it's better to make our way of figuring stuff out less basic alternatively ;)

**Comment by lantw44** [comment]
> how does CMake determine these paths ?

LLVM ships a config-file for CMake. Surprisingly, CMake finds config-files in a lot of places. It is not a single directory but a long list of directories with wildcards in them.
https://cmake.org/cmake/help/v3.15/command/find_package.html#search-procedure

On FreeBSD, the location matches with the pattern.
```
$ ls -l /usr/local/llvm*/lib/cmake/llvm*/LLVMConfig.cmake
-rw-r--r--  1 root  wheel  24543  1月 19 11:11 /usr/local/llvm60/lib/cmake/llvm/LLVMConfig.cmake
-rw-r--r--  1 root  wheel  25013  6月  2 03:12 /usr/local/llvm70/lib/cmake/llvm/LLVMConfig.cmake
-rw-r--r--  1 root  wheel  26999  6月  6 04:21 /usr/local/llvm80/lib/cmake/llvm/LLVMConfig.cmake
```

However, Meson currently doesn't tell CMake to choose the latest version, so it may pick the oldest one instead. https://github.com/mesonbuild/meson/issues/5636

**Comment by MathieuDuponchelle** [comment]
I see. In any case, exposing a way to specify the paths to llvm libs and headers seems necessary, so feel free to propose a patch that implements that @lantw44 :) If you also feel like improving the detection of llvm-config so that hotdoc works on its own as well that would be appreciated, but not required.
